test/recipes/05-test_rand.t

```diff
@@ -34,11 +35,12 @@ SKIP: {
     my @randdata;
     my $expected = '0102030405060708090a0b0c0d0e0f10';
 
-    @randdata = run(app(['openssl', 'rand', '-engine', 'ossltest', '-hex', '16' ]),
+    $ENV{OPENSSL_MODULES} = abs_path(bldtop_dir("test"));
+    @randdata = run(app(['openssl', 'rand', '-provider', 'p_ossltest', '-provider', 'default', '-propquery', '?provider=p_ossltest', '-hex', '16' ]),
```

Comment:
why the default provider is needed here?

Reply:
I assume its more related to it using the -propquery.